gfx/drivers/gl_core.c

```diff
@@ -0,0 +1,2001 @@
+   if (!gl->pipelines.font)
+      return false;
+
+   gl->pipelines.ribbon_simple = gl_core_cross_compile_program(pipeline_ribbon_simple_vert, sizeof(pipeline_ribbon_simple_vert),
+                                                               pipeline_ribbon_simple_frag, sizeof(pipeline_ribbon_simple_frag),
+                                                               &gl->pipelines.ribbon_simple_loc, true);
+   if (!gl->pipelines.ribbon_simple)
+      return false;
+
+   gl->pipelines.ribbon = gl_core_cross_compile_program(pipeline_ribbon_vert, sizeof(pipeline_ribbon_vert),
+                                                        pipeline_ribbon_frag, sizeof(pipeline_ribbon_frag),
+                                                        &gl->pipelines.ribbon_loc, true);
+   if (!gl->pipelines.ribbon)
+      return false;
+
+   gl->pipelines.bokeh = gl_core_cross_compile_program(alpha_blend_vert, sizeof(alpha_blend_vert),
+                                                       pipeline_bokeh_frag, sizeof(pipeline_bokeh_frag),
+                                                       &gl->pipelines.bokeh_loc, true);
+   if (!gl->pipelines.bokeh)
+      return false;
+
+   gl->pipelines.snow_simple = gl_core_cross_compile_program(alpha_blend_vert, sizeof(alpha_blend_vert),
+                                                             pipeline_snow_simple_frag, sizeof(pipeline_snow_simple_frag),
+                                                             &gl->pipelines.snow_simple_loc, true);
+   if (!gl->pipelines.snow_simple)
+      return false;
+
+   gl->pipelines.snow = gl_core_cross_compile_program(alpha_blend_vert, sizeof(alpha_blend_vert),
+                                                      pipeline_snow_frag, sizeof(pipeline_snow_frag),
+                                                      &gl->pipelines.snow_loc, true);
+   if (!gl->pipelines.snow)
+      return false;
+
+   return true;
+}
+
+static bool gl_core_init_default_filter_chain(gl_core_t *gl)
+{
+   if (!gl->ctx_driver)
+      return false;
+
+   gl->filter_chain = gl_core_filter_chain_create_default(
+         gl->video_info.smooth ?
+         GL_CORE_FILTER_CHAIN_LINEAR : GL_CORE_FILTER_CHAIN_NEAREST);
+
+   if (!gl->filter_chain)
+   {
+      RARCH_ERR("Failed to create filter chain.\n");
+      return false;
+   }
+
+   return true;
+}
+
+static bool gl_core_init_filter_chain_preset(gl_core_t *gl, const char *shader_path)
+{
+   gl->filter_chain = gl_core_filter_chain_create_from_preset(
+         shader_path,
+         gl->video_info.smooth ?
+         GL_CORE_FILTER_CHAIN_LINEAR : GL_CORE_FILTER_CHAIN_NEAREST);
+
+   if (!gl->filter_chain)
+   {
+      RARCH_ERR("[GLCore]: Failed to create preset: \"%s\".\n", shader_path);
+      return false;
+   }
+
+   return true;
+}
+
+static bool gl_core_init_filter_chain(gl_core_t *gl)
+{
+   const char *shader_path = retroarch_get_shader_preset();
```

Comment:
on most drivers we check if the shader is supported.. just in case
seems safe here but on some instances we had crashes on others

example:
```C
      const char* ext = path_get_extension(retroarch_get_shader_preset());

      if (ext && !strncmp(ext, "slang", 5))
d3d11_gfx_set_shader(d3d11, RARCH_SHADER_SLANG, retroarch_get_shader_preset());
```

Seems we don't check in gl though 😆 
Not sure how important this is.

Reply:
That path is not hit unless it has checked for SLANG type.